spec/unit/berkshelf/downloader_spec.rb

```diff
@@ -59,6 +59,31 @@ module Berkshelf
         subject.try_download(source, name, version)
       end
 
+      context "supports location paths" do
+        before(:each) do
+          allow(source).to receive(:type) { :supermarket }
+          allow(source).to receive(:options) { { ssl: {} } }
+          allow(source).to receive(:uri_string).and_return("http://localhost:8081/repository/chef-proxy")
+          allow(remote_cookbook).to receive(:location_type) { :opscode }
+        end
+
+        let(:rest) { double("community-rest") }
+
+        it "that are relative and prepends the source URI for the download" do
+          allow(remote_cookbook).to receive(:location_path) { "/api/v1" }
+          expect(CommunityREST).to receive(:new).with("http://localhost:8081/repository/chef-proxy/api/v1", { ssl: {} }) { rest }
+          expect(rest).to receive(:download).with(name, version)
+          subject.try_download(source, name, version)
+        end
+
+        it "that are absolute and uses the given absolute URI" do
+          allow(remote_cookbook).to receive(:location_path) { "http://some.bucket.somewhere.example.com:8081/repository/chef-proxy/api/v1" }
+          expect(CommunityREST).to receive(:new).with("http://some.bucket.somewhere.example.com:8081/repository/chef-proxy/api/v1", { ssl: {} }) { rest }
```

Comment:
I know I wrote this bit, but after looking at [`Downloader.try_download`](https://github.com/berkshelf/berkshelf/blob/master/lib/berkshelf/downloader.rb#L60) and [`CommunityREST.download`](https://github.com/berkshelf/berkshelf/blob/master/lib/berkshelf/community_rest.rb#L91), the `some.bucket.somewhere.example.com` I added here doesn't make sense. 🤦‍♂️

`Downloader.try_download` under test here is still poking at a particular Supermarket API endpoint. It's not until `CommunityREST.download` goes to [retrieve the actual archive](https://github.com/berkshelf/berkshelf/blob/master/lib/berkshelf/community_rest.rb#L92-L108) that a URL might be something other than the Supermarket endpoint (e.g. an S3 bucket for the actual cookbook artifact).

So, uh, `some.bucket.somewhere.example.com` can go back to being `localhost` to make sense in context. 😬

Reply:
In: 80977c3